README.md

```diff
@@ -1,66 +1,145 @@
+
+### Snapshot controller command line options
+
+#### Important optional arguments that are highly recommended to be used
+* `--leader-election`: Enables leader election. This is useful when there are multiple replicas of the same snapshot controller running for the same Kubernetes deployment. Only one of them may be active (=leader). A new leader will be re-elected when current leader dies or becomes unresponsive for ~15 seconds.
+
+* `--leader-election-namespace <namespace>`: The namespace where the leader election resource exists. Defaults to the pod namespace if not set.
+
+#### Other recognized arguments
+* `--kubeconfig <path>`: Path to Kubernetes client configuration that the snapshot controller uses to connect to Kubernetes API server. When omitted, default token provided by Kubernetes will be used. This option is useful only when the snapshot controller does not run as a Kubernetes pod, e.g. for debugging.
+
+* `--resync-period <duration>`: Internal resync interval when the snapshot controller re-evaluates all existing `VolumeSnapshot` instances and tries to fulfill them, i.e. create / delete corresponding snapshots. It does not affect re-tries of failed calls! It should be used only when there is a bug in Kubernetes watch logic.
+
+* `--create-snapshotcontent-retrycount`: Number of retries when we create a snapshot content object for a snapshot. Default is 5.
+
+* `--create-snapshotcontent-interval`: Interval between retries when we create a snapshot content object for a snapshot. Default is 10 seconds.
+
+* `--version`: Prints current snapshot controller version and quits.
+
+* All glog / klog arguments are supported, such as `-v <log level>` or `-alsologtostderr`.
+
+### CSI external snapshotter sidecar command line options
+
+#### Important optional arguments that are highly recommended to be used
+* `--csi-address <path to CSI socket>`: This is the path to the CSI driver socket inside the pod that the external-snapshotter container will use to issue CSI operations (`/run/csi/socket` is used by default).
+
+* `--leader-election`: Enables leader election. This is useful when there are multiple replicas of the same external-snapshotter running for one CSI driver. Only one of them may be active (=leader). A new leader will be re-elected when current leader dies or becomes unresponsive for ~15 seconds.
+
+* `--leader-election-namespace <namespace>`: The namespace where the leader election resource exists. Defaults to the pod namespace if not set.
+
+* `--timeout <duration>`: Timeout of all calls to CSI driver. It should be set to value that accommodates majority of `CreateSnapshot`, `DeleteSnapshot, and `ListSnapshots` calls. 1 minute is used by default.
+
+* `snapshot-name-prefix`: Prefix to apply to the name of a created snapshot. Default is `snapshot`.
+
+* `snapshot-name-uuid-length`: Length in characters for the generated uuid of a created snapshot. Defaults behavior is to NOT truncate.
```

Comment:
Do we make sure the length doesn't exceed the CSI spec max? especially when prefix is added?

Reply:
The CSI spec says length of the name conforms to the length limit of an unicode string.  The code we have to handle the snapshot name in the snapshotter is similar to how the volume name is handled in the provisioner.

https://github.com/kubernetes-csi/external-provisioner/blob/master/pkg/controller/controller.go#L319
https://github.com/kubernetes-csi/external-provisioner/blob/master/README.md#other-recognized-arguments

Reply:
CSI Spec says 128 max: https://github.com/container-storage-interface/spec/blob/master/spec.md#size-limits.

Let's followup on this as a separate issue?

Reply:
Sure

Reply:
Opened an issue here: https://github.com/kubernetes-csi/external-snapshotter/issues/216